troubleshoot-tidb-oom.md

```diff
@@ -0,0 +1,198 @@
+    - If OOM is caused by a database issue, the following are some possible causes:
+        - TiDB handles large data traffic, such as large queries, large writes, and data import.
+        - TiDB is in a high concurrency scenario, where multiple SQL concurrency consumes resources or operator concurrency is high.
+        - TiDB has a memory leak and resources are not freed.
+
+    Refer to the following sections for specific troubleshooting methods.
+
+## Typical causes and solutions
+
+Depending on the reasons for OOM issues, it can be divided into the following scenarios.
+
+- [Deployment issues](#deployment-issues)
+- [Database issues](#database-issues)
+- [Client side issues](#client-side-issues)
+
+### Deployment issues
+
+The following are common causes of OOM caused by deployment issues:
+
+- The memory capacity of the operating system is too small.
+- The TiUP configuration [`resource_control`](/tiup/tiup-cluster-topology-reference.md#global) is not appropriate.
+- In the case of hybrid deployments (meaning that TiDB and other applications are deployed on the same server), TiDB is killed accidentally by `oom-killer` due to lack of resources.
+
+### Database issues
+
+This section describes the causes and solutions for OOM caused by database issues.
+
+> **Note:**
+>
+> If you have configured [`tidb_mem_quota_query`](/system-variables.md#tidb_mem_quota_query), an error occurs: `ERROR 1105 (HY000): Out Of Memory Quota![conn_id=54]`. It is caused by the memory usage control behavior of the database. You can safely ignore it.
+
+#### Executing SQL statements consumes too much memory
+
+You can take the following measures to reduce the memory usage of SQL statements, depending on the different causes of OOM issues.
+
+- If the execution plan of SQL is not optimal, for example, due to lack of proper indexes, outdated statistics, or optimizer bugs, a wrong execution plan of SQL might be selected. A huge intermediate result set will then be accumulated in the memory. In this case, consider the following measures.
+    - Add appropriate indexes
+    - Use the [disk spill](/configure-memory-usage.md#disk-spill) feature for execution operators
+    - Adjust the JOIN order between tables
+    - Use hints to optimize SQL statements
+
+- Some operators and functions do not support pushing down to the storage level, resulting in a huge accumulation of intermediate result sets. In this case, you need to refine the business SQL statements or use hints to tune, and use the functions or operators that support pushing down.
+
+- The execution plan contains the operator HashAgg. HashAgg is executed concurrently by multiple threads, which is faster but consumes more memory. Instead, you can use `STREAM_AGG()`.
+
+- Reduce the number of regions to be read simultaneously or reduce the concurrency of operators to avoid memory problems caused by high concurrency. The corresponding system variables include:
```

Comment:
```suggestion
- Reduce the number of Regions to be read simultaneously or reduce the concurrency of operators to avoid memory problems caused by high concurrency. The corresponding system variables include:
```